common/src/main/java/org/dash/wallet/common/services/SecurityModel.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.common.services
+
+import androidx.fragment.app.FragmentActivity
+
+interface SecurityModel {
+    suspend fun requestPinCode(activity: FragmentActivity): String?
```

Comment:
`CheckPinDialog` is a bit heavy on wallet dependencies, so I inject this functionality with DI instead.